src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -827,17 +827,18 @@ public INamedTypeSymbol GetTypeByMetadataName(string fullyQualifiedMetadataName)
         protected abstract INamedTypeSymbol CommonGetTypeByMetadataName(string metadataName);
 
         /// <summary>
-        /// Returns a new INamedTypeSymbol with the given element types and (optional) element names.
+        /// Returns a new INamedTypeSymbol with the given underlying type and (optional) element names.
         /// </summary>
         /// <remarks>
         /// Since VB doesn't support tuples yet, this call will fail in a VB compilation.
+        /// Also, the underlying type needs to be tuple-compatible.
```

Comment:
For a tuple of cardinality 2 to 7 it should be ValueTuple of arity 2 to 7 accordingly. For a tuple of 8+ it should be a chain of one or more ValueTuple of arity 8 followed by a ValueTuple of arity 1 to 7. The chaining is done through type argument corresponding to TRest type parameter (the last type parameter) of ValueTuple of arity 8.  

---
In reply to: [62423814](https://github.com/dotnet/roslyn/pull/11160#discussion_r62423814) [](ancestors = 62423814)